rclcpp/include/rclcpp/timer.hpp

```diff
@@ -190,6 +201,39 @@ class GenericTimer : public TimerBase
   RCLCPP_DISABLE_COPY(GenericTimer)
 
   FunctorT callback_;
+
+private:
+  template<
+    typename CallbackT = FunctorT,
+    typename std::enable_if<
+      rclcpp::function_traits::same_arguments<CallbackT, VoidCallbackType>::value
+    >::type * = nullptr
+  >
+  void
+  register_callback_function()
```

Comment:
@iluetkeb same nit about naming, though in this case maybe `register_callback_tracepoint()` makes more sense?

Reply:
Well, as I understand it, a "tracepoint" is not a location in the user code, but a tracing function with a specific signature. This seems slightly counter-intuitive in naming, so @christophebedard correct me if I'm wrong, but it appears to be used like that by the trace libraries.

This means, the tracepoint is already there and what's being registered is the user callback function within the ROS node.

So, I would say the naming is appropriate.

Reply:
> It's not immediately obvious that you're registering the symbol of a callback for tracing.

I agree. To be honest, it was a name I used at first to try doing it as a workaround and it just stuck.

Maybe something like `register_callback_symbol()`? Maybe it's still not clear that it's referring to the callback _function_. I'm open to anything obviously (for all instances of this registration pattern).

Reply:
Maybe `register_callback_for_tracing()`?  Just to associate this functionality with its final purpose.